src/accountsdb/db.zig

```diff
@@ -2147,17 +2147,15 @@ pub const AccountsDB = struct {
 
         const slot_entry = try slot_ref_map.getOrPut(slot);
 
-        if (!slot_entry.found_existing) {
-            // no entry means realloc always needed, value set in realloc_needed block
-            slot_entry.value_ptr.* = .{
-                .refs = .empty,
-                .global_index = std.math.maxInt(u64), // u64-max == invalid value / replaced soon
-            };
```

Comment:
Is it safe to remove this? what if this function returns with an error before setting the value?

Reply:
If the function errors before setting the value, the old refs will continue to be accessible+valid - it's safer to not have this

Reply:
Oh I misunderstood, I see the problem.

Opted to implement both guards - 01a49c82d487160020064041d0f4bc0c68a61da5. I think removing from the index in an errdefer makes the default-initialize redundant? But it's much easier to just have both than to think about it right now.

Reply:
I think the default initialize is easier to reason about because what you're doing is making sure that the state is always correct at every step of execution.

I find the errdefer a little harder to reason about because it leaves the state incorrect for longer and then tries to roll it back if necessary. I think it would have the correct effect, but maybe it's less resistant to future refactoring bugs being introduced.